applications/MeshingApplication/custom_processes/metrics_error_process.cpp

```diff
@@ -0,0 +1,248 @@
+
+    /******************************************************************************
+    --3-- Calculate metric (for each node) --3--
+    ******************************************************************************/
+    CalculateMetric();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void MetricErrorProcess<TDim>::CalculateElementSize()
+{
+    // Get error values
+    const ProcessInfo& r_process_info = mrThisModelPart.GetProcessInfo();
+    const double energy_norm_overall = r_process_info[ENERGY_NORM_OVERALL];
+    const double error_overall = r_process_info[ERROR_OVERALL];
+
+    // Some initializations
+    const double tolerance = std::numeric_limits<double>::epsilon();
+
+     // Loop over all elements:
+    ElementsArrayType& elements_array = mrThisModelPart.Elements();
+    const int num_elem = static_cast<int>(elements_array.size());
+
+    // Compute new element size
+    #pragma omp parallel for
+    for(int i_elem = 0; i_elem < num_elem; ++i_elem){
+        auto it_elem = elements_array.begin() + i_elem;
+
+        //Compute the current element size h
+        ComputeElementSize(it_elem);
+
+        // Compute new element size
+        const double element_error = it_elem->GetValue(ELEMENT_ERROR);
+        const double coeff = std::abs(element_error) < tolerance ? 1.0 : 1.0/element_error;
+        double new_element_size = coeff * it_elem->GetValue(ELEMENT_H);
+
+        // If a target number for elements is given: use this, else: use current element number
+        // if(mSetElementNumber == true && mElementNumber<mrThisModelPart.Elements().size())
+        if(mSetElementNumber == true)
+            new_element_size *= std::sqrt((std::pow(energy_norm_overall, 2) + std::pow(error_overall, 2))/mElementNumber) * mTargetError;
+        else
+            new_element_size *= std::sqrt((energy_norm_overall*energy_norm_overall+error_overall*error_overall)/mrThisModelPart.Elements().size())*mTargetError;
+
+
+        // Check if element sizes are in specified limits. If not, set them to the limit case
+        if(new_element_size < mMinSize)
+            new_element_size = mMinSize;
+        if(new_element_size > mMaxSize)
+            new_element_size = mMaxSize;
+
+        it_elem->SetValue(ELEMENT_H, new_element_size);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void MetricErrorProcess<TDim>::CalculateMetric()
+{
+    // Array of nodes
+    NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+
+    // We do a find of neighbours
+    FindNodalNeighboursProcess find_neighbours(mrThisModelPart);
+    if (nodes_array.begin()->Has(NEIGHBOUR_ELEMENTS)) find_neighbours.ClearNeighbours();
```

Comment:
What?

Reply:
{
         FindNodalNeighboursProcess find_neighbours(mrThisModelPart);
         if (nodes_array.begin()->Has(NEIGHBOUR_ELEMENTS)) find_neighbours.ClearNeighbours();
         ...
    }

well this is better (my previous suggestion is not correct). I just put a {} around the code block with the process so that the process is deleted outside of that scope (it is not needed any longer).

feel free to forget this comment if you prefer

Reply:
Do you mean this?, OK

Reply:
Done